src/lib.rs

```diff
@@ -2000,6 +2000,21 @@ pub trait Itertools: Iterator {
     /// assert_eq!(numbers.iter().find_or_last(|&&x| x > 2), Some(&3));
     /// assert_eq!(std::iter::empty::<i32>().find_or_last(|&x| x > 5), None);
     /// ```
+    ///
+    /// An iterator of [`Result`]s can find the first `Ok` or the last `Err`:
```

Comment:
I think we can make it less prominent. Maybe just add the following after the previous `assert_eq`s:

```
// find first Ok if present, or return the last Err:
assert_eq!(vec![Err(1), Ok(2), Err(3), Ok(4)].find_or_last(Result::is_ok), Ok(2));
assert_eq!(vec![Err(1), Err(2)].find_or_last(Result::is_ok), Err(2));
...
```